src/callbacks_step/amr.jl

```diff
@@ -415,12 +415,27 @@ function (amr_callback::AMRCallback)(u_ode::AbstractVector, mesh::P4estMesh,
     coarsened_original_cells = Int[]
   end
 
-  # Store whether there were any cells coarsened or refined
+  # Store whether there were any cells coarsened or refined and perform load balancing
   has_changed = !isempty(refined_original_cells) || !isempty(coarsened_original_cells)
+  # Check if mesh changed on other processes
+  if mpi_isparallel()
+    has_changed = MPI.Allreduce!(Ref(has_changed), |, mpi_comm())[]
+  end
+
   if has_changed # TODO: Taal decide, where shall we set this?
     # don't set it to has_changed since there can be changes from earlier calls
     mesh.unsaved_changes = true
 
+    if mpi_isparallel() && amr_callback.dynamic_load_balancing
+      @trixi_timeit timer() "dynamic load balancing" begin
+        global_first_quadrant = unsafe_wrap(Array, mesh.p4est.global_first_quadrant, mpi_nranks() + 1)
+        old_global_first_quadrant = Vector{eltype(global_first_quadrant)}(undef, mpi_nranks() + 1)
+        old_global_first_quadrant .= global_first_quadrant
```

Comment:
Does
```suggestion
        global_first_quadrant = unsafe_wrap(Array, mesh.p4est.global_first_quadrant, mpi_nranks() + 1)
        old_global_first_quadrant = copy(global_first_quadrant)
```
work, too?

Reply:
Hm, I think it should work. Would it be necessary to `GC.@preserve` anything here? I'm always a bit cautious when dealing with `unsafe` stuff.

Reply:
No, it shouldn't. It would be problematic if `mesh.p4est.global_first_quadrant` was invalidated during the lifetime of `global_first_quadrant`, but that's also the case for the current version above.

Reply:
If the simpler version `old_global_first_quadrant = copy(global_first_quadrant)` does the same thing, I prefer it. Otherwise, please add a comment explaining why it doesn't work.

Reply:
The simpler version works fine, I changed it accordingly in [9adb087](https://github.com/trixi-framework/Trixi.jl/pull/1080/commits/9adb087e2b6e2f8889182d415c4115ca53072ce2).